### PR TITLE
 expression: implement arithmetic vec int divide int sig

### PR DESCRIPTION
**Pull request by ZiheLiu**
PCP #12103

### What problem does this PR solve? 
implement builtinArithmeticIntDivideIntSig vectorized function
#12103

### What is changed and how it works?
- implement vectorized builtinArithmeticIntDivideIntSig
- add unit test for vectorized builtinArithmeticIntDivideIntSig with 4 test cases:
    - int64 divides int64
    - uint64 divides uint64
    - int64 divides uint64
    - uint64 divides int64

about 3 times faster
```shell
$  go test -v -benchmem -bench=BenchmarkVectorizedBuiltinArithmeticFunc -run=BenchmarkVectorizedBuiltinArithmeticFunc -args builtinArithmeticIntDivideIntSig
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-VecBuiltinFunc-8                 64203             18457 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-NonVecBuiltinFunc-8              25732             46684 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-VecBuiltinFunc#01-8             106828             11296 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-NonVecBuiltinFunc#01-8           28982             40606 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-VecBuiltinFunc#02-8              78523             15427 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-NonVecBuiltinFunc#02-8           26662             44313 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-VecBuiltinFunc#03-8              78825             15292 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-NonVecBuiltinFunc#03-8           27369             44987 ns/op               0 B/op          0 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      12.083s

```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by qw4990** [comment]
It panicked, and here is the stack:
```
[2020-01-19T06:33:26.465Z] PANIC: builtin_arithmetic_vec_test.go:182: testEvaluatorSuite.TestVectorizedBuiltinArithmeticFunc
[2020-01-19T06:33:26.465Z] 
[2020-01-19T06:33:26.465Z] ... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0xEA6C21)
[2020-01-19T06:33:26.465Z] 
[2020-01-19T06:33:26.465Z] /usr/local/go/src/runtime/panic.go:679
[2020-01-19T06:33:26.465Z]   in gopanic
[2020-01-19T06:33:26.465Z] /usr/local/go/src/runtime/panic.go:199
[2020-01-19T06:33:26.465Z]   in panicmem
[2020-01-19T06:33:26.465Z] /usr/local/go/src/runtime/signal_unix.go:394
[2020-01-19T06:33:26.465Z]   in sigpanic
[2020-01-19T06:33:26.465Z] bench_test.go:533
[2020-01-19T06:33:26.465Z]   in rangeInt64Gener.gen
[2020-01-19T06:33:26.465Z] bench_test.go:958
[2020-01-19T06:33:26.465Z]   in fillColumnWithGener
[2020-01-19T06:33:26.465Z] bench_test.go:946
[2020-01-19T06:33:26.465Z]   in fillColumn
[2020-01-19T06:33:26.465Z] bench_test.go:1174
[2020-01-19T06:33:26.465Z]   in genVecBuiltinFuncBenchCase
[2020-01-19T06:33:26.465Z] bench_test.go:1279
[2020-01-19T06:33:26.465Z]   in testVectorizedBuiltinFunc
[2020-01-19T06:33:26.465Z] builtin_arithmetic_vec_test.go:183
[2020-01-19T06:33:26.465Z]   in testEvaluatorSuite.TestVectorizedBuiltinArithmeticFunc
[2020-01-19T06:33:26.465Z] /usr/local/go/src/reflect/value.go:321
[2020-01-19T06:33:26.465Z]   in Value.Call
[2020-01-19T06:33:26.465Z] /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/pkg/mod/github.com/pingcap/check@v0.0.0-20191216031241-8a5a85928f12/check.go:850
[2020-01-19T06:33:26.465Z]   in suiteRunner.forkTest.func1
[2020-01-19T06:33:26.465Z] /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/pkg/mod/github.com/pingcap/check@v0.0.0-20191216031241-8a5a85928f12/check.go:739
[2020-01-19T06:33:26.465Z]   in suiteRunner.forkCall.func1
[2020-01-19T06:33:26.465Z] /usr/local/go/src/runtime/asm_amd64.s:1357
[2020-01-19T06:33:26.465Z]   in goexit
```
@ZiheLiu Please fix this problem at your convenience.

File: expression/builtin_arithmetic_vec.go [hunk 1]
```diff
@@ -814,11 +813,136 @@ func (b *builtinArithmeticDivideRealSig) vecEvalReal(input *chunk.Chunk, result
...
+	}
+
+	// reuse result as rh to avoid buf allocate
+	if err := b.args[1].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	result.MergeNulls(lh)
+
+	rh := result
```
Comment (zz-jason) [inline expression/builtin_arithmetic_vec.go]: it's better to `s/lh/lhs/` and `s/rh/rhs/`, for `lhs` is short for `left hand side` and `rhs` is short for `right hand side`

Comment (ZiheLiu) [reply expression/builtin_arithmetic_vec.go]: Thanks for your review. Should I rename `lh` to `lhs` and `rh` to `rhs`?

Comment (zz-jason) [reply expression/builtin_arithmetic_vec.go]: yes

**Pull request merged**